File: Pascal_Tree.py
def faktoriyel(i):
    fakti=1
    for k in range(1,i+1):
        fakti*=k
        
    return fakti
        
def sayiac(i,j):
    acilansayi = i
    for k in range(1,j):
        if i > 1:
            i-=1
        acilansayi*=i
    return acilansayi
       
    
    
sayi = int(input("Sayi Gir: "))
print(sayi)
bosluk = sayi*2   
for i in range(sayi+1):
    for j in range(bosluk):
        print(" ",end='')
    bosluk-=1
    for j in range(i+1):
        if j == 0 or j == i:
            print(1,"",end='')
        else:
            print(int(sayiac(i,j)/faktoriyel(j)),end=' ')
    print()
        

File: Pascal_Tree.cpp
#include <iostream>
using namespace std;
class pascal_tree{
    public:
        int faktori(int j){
            int fakti=1;
            for (int i = 1; i < j+1; i++)
            {
                fakti*=i;
            }
            return fakti;
        }

        int sayiac(int i,int j){
            int acilmis=i;
            for (int k = 1; k < j; k++)
            {
                if (i>1)
                {
                    i--;
                }
                acilmis*=i;
                
            }
            return acilmis;
        }

};


int main(){
    pascal_tree pascal_tree;
    int sayi,kadar=10;
    cout << "Sayi giriniz: ";                                             
    cin >> sayi;
    for (int j = 0; j < sayi; j++)
    {
        for (int b = kadar; b > 0; b--)
            {
                cout << " ";
            }
        kadar--;
        for (int i = 0; i <= j; i++)
        {
            if (i == 0 || i== j)
            {
                cout << 1 << " ";
            }
            else{
                cout << pascal_tree.sayiac(j,i) / pascal_tree.faktori(i) << " ";
            }
        }
        cout << "\n";
    }
    return 0;
}
  

  